dev/sg/internal/sgconf/global.go

```diff
@@ -0,0 +1,82 @@
+package sgconf
+
+import (
+	"os"
+	"path/filepath"
+	"sync"
+
+	"github.com/sourcegraph/sourcegraph/dev/sg/root"
+	"github.com/sourcegraph/sourcegraph/lib/output"
+)
+
+const (
+	DefaultFile          = "sg.config.yaml"
+	DefaultOverwriteFile = "sg.config.overwrite.yaml"
+)
+
+var (
+	globalConfOnce sync.Once
+	globalConf     *Config
+	globalConfErr  output.FancyLine
+)
+
+// Get retrieves the global config files. If Config is nil, a line will be provided that
+// can be printed for an explanation as to what went wrong.
+//
+// It must not be called before flag initalization, i.e. when confFile or overwriteFile is
+// not set.
+func Get(confFile, overwriteFile string) (*Config, output.FancyLine) {
```

Comment:
Can we change this to this?

```suggestion
func Get(confFile, overwriteFile string) (*Config, error) {
```

Because it seems like the majority of callers ignore the second arg and instead have to do a non-nil check on `*Config`. If we follow the Go convention it's clear that `if err == nil` then `*Config` would be non-nil.

Reply:
Sure! I kept it as fancyline because the messages have pretty nice formatting at the moment, but a better interface probably trumps that 👍